src/vs/platform/terminal/node/terminalEnvironment.ts

```diff
@@ -284,16 +288,19 @@ export function getShellIntegrationInjection(
 }
 
 /**
- * On macOS the profile calls path_helper which adds a bunch of standard bin directories to the
- * beginning of the PATH. This causes significant problems for the environment variable
+ * There are a few situations where some directories are added to the beginning of the PATH.
+ * 1. On macOS when the profile calls path_helper.
+ * 2. On fish when it automatically prepends "$fish_user_paths" to the PATH.
+ *
+ * This causes significant problems for the environment variable
  * collection API as the custom paths added to the end will now be somewhere in the middle of
  * the PATH. To combat this, VSCODE_PATH_PREFIX is used to re-apply any prefix after the profile
  * has run. This will cause duplication in the PATH but should fix the issue.
  *
  * See #99878 for more information.
  */
-function addEnvMixinPathPrefix(options: ITerminalProcessOptions, envMixin: IProcessEnvironment): void {
-	if (isMacintosh && options.environmentVariableCollections) {
+function addEnvMixinPathPrefix(options: ITerminalProcessOptions, envMixin: IProcessEnvironment, shell: string): void {
+	if ((isMacintosh || (isLinux && shell === 'fish')) && options.environmentVariableCollections) {
```

Comment:
> While further testing it, I noticed that the envMixin should be applied even on non-login shells with fish, because the variable $fish_user_paths is always set

If this is the case, should this instead be `isMacintosh || shell === 'fish'` and not mention Linux?